src/ORM/DataObject.php

```diff
@@ -1257,7 +1257,7 @@ public function forceChange()
      */
     public function validate(): ValidationResult
     {
-        $result = ValidationResult::create();
+        $result = ValidationResult::create(get_class($this), $this->ID);
```

Comment:
Is this the only place this gets set in core right now?

What about if I call `validate()` on a form, and the form belongs to a specific model? E.g. I could conceivably set up a PHP `Form` for some record that can be submitted from CLI using the fancy symfony console IO functionality.
Personally I think that scenario is likely to contain all the relevant context already, but want to get your thoughts and make sure that sort of thing has been considered.

Reply:
A Form has no awareness of a DataObject, it simply keeps track of key=value style data.  The only time it interacts with DataObject is when calling Form::saveInto(DataObjectInterface $dataObject)

It's the same story with most the other things that call ValidationResult::create() and could be related to DataObjects, e.g. FormField, ConstraintValidator, FieldValidator

MemberAuthenticator and PasswordValidator do deal with Member DataObjects, though I don't we shouldn't be displaying any values for those in a CLI context

Reply:
I've updated the PR to remove the constructor arguments as that functionality is basically only for DataObjects and not for other thing contexts ValidationResult is used in such as Forms. I've updated DataObject::validate() to call `$result->setModelClass(static::class);` and `$result->setRecordID($this->ID);` instead

Reply:
> I've updated the PR to remove the constructor arguments as that functionality is basically only for DataObjects 

I think the additional context should still be settable in the constructor. It's useful for any scenario in a project where you're setting validation results on any model and you don't expect the result to be displayed in a normal CMS edit context. i.e. projects can use this the same way we're using it in `DataObject`.

That said, the setters are public so we're not locking people out of that, just making the functionality a little harder to discover - so I won't block merging if that isn't done.

Reply:
> A Form has no awareness of a DataObject

Form has a `$record` property which gets populated when `$form->loadDataFrom()` is called.

> MemberAuthenticator and PasswordValidator do deal with Member DataObjects, though I don't we shouldn't be displaying any values for those in a CLI context

I assume you meant "I think we shouldn't" - in which case I agree, but none of this functionality deals with values. It deals with model classes and IDs.

Reply:
I'd like to keep the constructor without arguments. I don't think there's any further changes required here

Reply:
`FormRequestHandler::httpSubmission()` is called _after_ the form has been instantiated - and instantiating the form may include calling `loadDataFrom($record)` (e.g. see `LeftAndMain::getEditForm()`, `AssetFormFactory::getForm()`, and [our documentation](https://docs.silverstripe.org/en/5/developer_guides/forms/how_tos/handle_nested_data/#form-fields)).

Reply:
Since it's a edge case I don't think we need to worry about it.

Remember the original motivation for this work was to help debug any new validation exceptions that suddently popped out when running dev/build when people migrated to CMS 6 now that we've added DBField validation.